bundles/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/handler/OpenSprinklerDeviceHandler.java

```diff
@@ -57,4 +72,29 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         // nothing to do here
     }
 
+    @Override
+    public void updateChannels() {
```

Comment:
Is the current sensor something that can be added or removed in realtime or is it something intrinsic to the device itself? If it can't be changed in realtime it would be better to add the channel during handler initialization so it gets executed only once rather than every time channel values are updated.

Reply:
It looks like the design pattern you are using is rather common in other bindings, so disregard this change request.

Reply:
Where? IMO this needs to be changed. Thing updates should only be done if the thing really needs chaning. I think in this case it should be done in `initialize()`.

Reply:
Done